client/state/plugins/wporg/reducer.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import React from 'react/addons';
+
+function updatePluginState( state = {}, pluginSlug, attributes ) {
+	return Object.assign( {},
+		state,
+		{ [ pluginSlug ]: Object.assign( {}, state[ pluginSlug ], attributes ) }
+	);
+}
```

Comment:
Shouldn't we actually even try to deprecate React's own immutability helpers, for consistency's sake (and stick with either `Object.assign()`, or Immutable.js)?